lib/langchain/active_record/hooks.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module Langchain
+  module ActiveRecord
+    #
+    # This module adds the following functionality to your ActiveRecord models:
+    # * `vectorsearch` class method to set the provider
+    # * `similarity_search` class method to search for similar texts
+    # * `index_to_vectorsearch` instance method to index the text to the provider
+    #
+    # Usage:
+    #     class Recipe < ActiveRecord::Base
+    #       vectorsearch provider: Langchain::Vectorsearch::Weaviate.new(
+    #                    api_key: ENV["WEAVIATE_API_KEY"],
+    #                    url: ENV["WEAVIATE_URL"],
+    #                    index_name: "Recipes",
```

Comment:
I'd like to see this handled automatically. 

I was thinking this could be handled in a similar way to `ActiveRecord::Base.table_name`? 
 - https://api.rubyonrails.org/classes/ActiveRecord/ModelSchema/ClassMethods.html#method-i-table_name-3D
 - https://api.rubyonrails.org/classes/ActiveRecord/ModelSchema/ClassMethods.html#method-i-table_name

Giving it more thought though, the index name is needed at the time vector search is instantiated. Could make it work if we made it lazily instantiated. Maybe something like...

```ruby
vectorsearch weaviate: {
  api_key: ENV["WEAVIATE_API_KEY"],
  url: ENV["WEAVIATE_URL"],
}

# then in the class method
def vectorsearch(providers = {})
  # only allow one to be passed in
  # providers.keys corresponds to the object
  provider = providers.keys.first
  self.class.class_variable_set(:@@provider_options, providers[provider))
  # ...
end

def vectorsearch_provider
  @vectorsearch_provider ||=  ... # lookup provider by name, pass in the options
end
```